src/main/java/org/takes/rs/RsPrettyXML.java

```diff
@@ -114,10 +117,44 @@ private Response make() throws IOException {
                 OutputKeys.OMIT_XML_DECLARATION, "yes"
             );
             transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+            if (isHtml5(source.getInputSource().getByteStream())) {
+                // @checkstyle MultipleStringLiteralsCheck (1 line)
+                transformer.setOutputProperty(OutputKeys.METHOD, "html");
+                transformer.setOutputProperty(OutputKeys.VERSION, "5.0");
+            }
             transformer.transform(source, new StreamResult(result));
         } catch (final TransformerException ex) {
             throw new IOException(ex);
         }
         return result.toByteArray();
     }
+
+    /**
+     * Checks if the input is an html5 document.
+     * @param body The body to be checked.
+     * @return True if the input is an html5 document.
+     * @checkstyle MethodNameCheck (3 lines)
+     */
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    private static boolean isHtml5(final InputStream body) {
+        Document document = null;
+        boolean valid = true;
+        try {
+            body.mark(Integer.MAX_VALUE);
+            final DocumentBuilderFactory factory =
+                DocumentBuilderFactory.newInstance();
+            final DocumentBuilder builder = factory.newDocumentBuilder();
+            document = builder.parse(body);
+            body.reset();
+            // @checkstyle IllegalCatchCheck (1 line)
+        } catch (final Exception ex) {
```

Comment:
@mbarbieri What do you mean it's not a "real exception"? Why not catch `SAXException` and `IOException` separately? Seems to me that they are thrown in different situations - see [javadoc](http://docs.oracle.com/javase/7/docs/api/javax/xml/parsers/DocumentBuilder.html#parse(java.io.InputStream)).